testsuite/single/tests/efuns/read_file.c

```diff
@@ -18,6 +18,12 @@ void do_tests() {
     ASSERT(all == read_file("/testfile", 10, 0x7fffffff));
     ASSERT(!read_file("/does_not_exist"));
     ASSERT(!read_file("/testfile", 10000, 1));
+    
+    // Start tests for end of file -lines
+    mid = implode(explode(foo, "\n")[<14..<5], "\n") + "\n";
+    ASSERT_EQ(mid, read_file("/testfile", -15, 10));
+    // if we start too many lines before the end of the file, start at the beginning
+    ASSERT(read_file("/testfile", -10000, 1));
```

Comment:
i think you want `ASSERT_EQ(implode(explode(foo, "\n")[0..0],  read_file(...) )`